build.sbt

```diff
@@ -57,7 +58,10 @@ lazy val coreJS = project
     name := "nspl-core-js",
     target := file("core/targetJS"),
     Compile / sourceManaged := (Compile / sourceManaged ).value.getAbsoluteFile,
-    libraryDependencies += "org.scala-lang.modules" %%% "scala-collection-compat" % "2.6.0"
+    libraryDependencies ++= Seq(
+      "org.scala-lang.modules" %%% "scala-collection-compat" % "2.7.0",
+      ("org.scala-js" %%% "scalajs-java-securerandom" % "1.0.0").cross(CrossVersion.for3Use2_13)
```

Comment:
Maybe it would be better to not depend on this and as you suggested create a simple, non-secure random id (with s.u.Random.nextLong).

I will see whether that is viable.